node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -614,6 +611,40 @@ impl<C: Context> ActiveValidator<C> {
     pub(crate) fn next_round_length(&self) -> TimeDiff {
         state::round_len(self.next_round_exp)
     }
+
+    /// Logs a message if the witness unit is in a round without a proposal.
+    fn log_if_missing_proposal(
```

Comment:
This will only fire if node is an active validator, is there a way to have similar functionality for read-only nodes?

Reply:
That's a good point.
We'd need another timer for that, then. But a read-only node doesn't even have its own round length — so would we always assume minimum round length and log every round without a proposal? With our current configuration that would probably make sense.

Reply:
I was rather thinking about inspecting the panorama of a unit and checking if it's in a round that doesn't have a proposal. I _think_ it should be doable:
* if fork choice would return a different block than that round's leader block, we can check if there was a leader block in the panorama
* if fork choice returns a block from the previous round and there's no new leader block in the panorama, then it would have meant that the rounds misses the leader's block.

Reply:
Yes, but we don't want to have 100 lines of lots for each missed round, i.e. we don't want to log it for _every_ incoming witness message in that round.